src/httpServer/server.ts

```diff
@@ -66,10 +66,26 @@ export class DBOSHttpServer {
   async listen(port: number) {
     try {
       await this.checkPortAvailability(port, "127.0.0.1");
+    } catch (error) {
+      const err = error as NodeJS.ErrnoException;
+      if (err.code === 'EADDRINUSE') {
+        this.logger.error(`Port ${port} is already used for IPv4 address "127.0.0.1". Please use the -p option to choose another port.\n${err.stack}`);
+        process.exit(1);
+      } else {
+        this.logger.warn(`Error occurred while checking port availability for IPv4 address "127.0.0.1" : ${err.code}\n${err.stack}`);
```

Comment:
Same comment here around passing the stack. as part of the error string. We will have duplicate stacks.

Reply:
See above example -- it is not duplicated because the error stack from node goes deeper.